gard_owl_ingest/mondo_mapping_status.py

```diff
@@ -96,61 +97,78 @@ def gard_mondo_mapping_status(
+    # todo: should everything for gard-mondo.sssom.tsv be its own function?
+    # - sssom_proxyable_df: GARD::proxy mappings. Native from GARD.
     rows = []
-    # todo: include: prefix_map etc as comment
-    # todo: include: cols: subject_label, mapping_justification
-    for s, xref_dict in mappings.items():
+    for s, xref_dict in gard_native_mappings.items():
         for p, o_list in xref_dict.items():
             for o in o_list:
                 rows.append({
                     'subject_id': s,
                     'predicate_id': p,
                     'object_id': o,
                 })
-    sssom_proxy_df = pd.DataFrame(rows)
-    sssom_proxy_df2 = pd.merge(
-        sssom_proxy_df, mondo_sssom_df, how='left', left_on='object_id', right_on='mondo_object_id').fillna('')\
+    sssom_proxyable_df = pd.DataFrame(rows)
+
+    # - sssom_proxy_df: GARD::Mondo JOINed mappings via proxy (Orphanet/OMIM)
+    proxy_and_unmapped_df = pd.merge(
+        sssom_proxyable_df, mondo_sssom_df, how='left', left_on='object_id', right_on='mondo_object_id').fillna('')\
         .sort_values(by='subject_id')
-    del sssom_proxy_df2['mondo_object_id']
-    sssom_proxy_df2['mapping_justification'] = 'semapv:ManualMappingCuration'
-    proxied_gard_ids = set(sssom_proxy_df['subject_id'].tolist())
-    sssom_proxy_df2['gard_mondo_mapping_type'] = sssom_proxy_df2.apply(
+    del proxy_and_unmapped_df['mondo_object_id']
+    proxy_and_unmapped_df['mapping_justification'] = 'semapv:ManualMappingCuration'
+    # - proxy_existing: These were mappings that were made from Mondo to GARD before. We're likely going to ignore these
+    # completely. But every term on this list also is able to have a new proxy mapping between GARD and Mondo using our
+    # new algorithmic methods here.
+    # - unmapped: It's possible there may have been some previous Mondo::GARD mappings that were mapped that are
+    # actually in this list, but we are ignoring those now.
+    proxy_and_unmapped_df['gard_mondo_mapping_type'] = proxy_and_unmapped_df.apply(
         lambda x: 'proxy_existing' if x.subject_id in existing_gard_mondo_mappings
         else 'proxy_new' if x.mondo_id != ''
-        else 'proxy_new_via_other_path' if x.subject_id in proxied_gard_ids
-        else 'unexpected_unclassified', axis=1)
-    # - Add missing mappings
-    unmapped_ids: List[CURIE] = [y for y in in_gard if y not in proxied_gard_ids]
-    rows = [{'subject_id': x, 'gard_mondo_mapping_type': 'unmapped'} for x in unmapped_ids]
-    missing_df = pd.DataFrame(rows)
+        else 'unmapped', axis=1)
+
+    # todo: temp until unit test
+    # - testing
+    test_proxy_unmapped_gard = set(
+        proxy_and_unmapped_df[proxy_and_unmapped_df['gard_mondo_mapping_type'] == 'unmapped']['object_id'])
+    test_proxy_unmapped_mondo = set(mondo_sssom_df['mondo_object_id'])
+    assert len(test_proxy_unmapped_mondo.intersection(test_proxy_unmapped_gard)) == 0
```

Comment:
I don't normally have asserts in my code, but for now, while I'm still wary of things, until my unit testing is more robust, I'm gonna leave this here.